sections/Deals/CreateDeal.tsx

```diff
@@ -264,6 +264,41 @@ const CreateDeal: FC<CreateDealProps> = ({ poolAddress, purchaseToken }) => {
 		};
 	}, [formik?.values, poolAddress, provider, signer, walletAddress]);
 
+	useEffect(() => {
+		const getUnderlyingDealTokenTotal = async () => {
+			if (
+				!formik.values?.purchaseTokenTotal ||
+				formik.values.dealExchangeRate <= 0 ||
+				!signer ||
+				!formik.values?.underlyingDealToken
+			) {
+				return wei(0);
+			}
+
+			const underlyingDealContract = new ethers.Contract(
+				formik.values?.underlyingDealToken,
+				erc20Abi,
+				provider
+			);
+
+			const underlyingDealTokenDecimals = await underlyingDealContract.decimals();
+
+			setUnderlyingDealTokenTotal(
+				wei(formik.values?.purchaseTokenTotal ?? 0, underlyingDealTokenDecimals).mul(
```

Comment:
also, we don't want the underlyingDealTokenDecimals here. This is for purchaseTokenDecimals